drivers/display/ssd1322.c

```diff
@@ -102,103 +103,160 @@ static int ssd1322_blanking_off(const struct device *dev)
-			} else {
-				buf_out[seg_idx / SEGMENTS_PER_BYTE] |= color << shift;
-			}
+	const struct ssd1322_config *config = dev->config;
+	int i = 0;
+
+	if (config->segments_per_pixel == 2) {
+		for (; i < config->conversion_buf_size && pixel_count > cur_offset + i; i += 1) {
+			config->conversion_buf[i] = (buf[cur_offset + i] >> 4) << 4;
+			config->conversion_buf[i] |= buf[cur_offset + i] >> 4;
+		}
+	} else {
+		for (; i / 2 < config->conversion_buf_size && pixel_count > cur_offset + i;
+		     i += 2) {
+			config->conversion_buf[i / 2] = buf[cur_offset + i + 1] >> 4;
+			config->conversion_buf[i / 2] |= (buf[cur_offset + i] >> 4) << 4;
 		}
 	}
+	return i;
+}
 
-	buf_in += pixels_in_chunk / 8;
-	*pixel_count -= pixels_in_chunk;
-	return pixels_in_chunk * segments_per_pixel / SEGMENTS_PER_BYTE;
+/* Convert what the conversion buffer can hold to pixelx (3:0) and pixelx+1 (7:4) */
+static int ssd1322_convert_MONO01(const struct device *dev, const uint8_t *buf, int cur_offset,
+				  uint32_t pixel_count)
+{
+	const struct ssd1322_config *config = dev->config;
+	int i = 0;
+
+	if (config->segments_per_pixel == 2) {
+		for (; i * 8 < config->conversion_buf_size && pixel_count > cur_offset + i;
+		     i += 1) {
+			config->conversion_buf[i * 8] = buf[cur_offset + i] & BIT(0) ? 0xFF : 0;
+			config->conversion_buf[i * 8 + 1] = buf[cur_offset + i] & BIT(1) ? 0xFF : 0;
+			config->conversion_buf[i * 8 + 2] = buf[cur_offset + i] & BIT(2) ? 0xFF : 0;
+			config->conversion_buf[i * 8 + 3] = buf[cur_offset + i] & BIT(3) ? 0xFF : 0;
+			config->conversion_buf[i * 8 + 4] = buf[cur_offset + i] & BIT(4) ? 0xFF : 0;
+			config->conversion_buf[i * 8 + 5] = buf[cur_offset + i] & BIT(5) ? 0xFF : 0;
+			config->conversion_buf[i * 8 + 6] = buf[cur_offset + i] & BIT(6) ? 0xFF : 0;
+			config->conversion_buf[i * 8 + 7] = buf[cur_offset + i] & BIT(7) ? 0xFF : 0;
+		}
+	} else {
+		for (; i * 4 < config->conversion_buf_size && pixel_count > cur_offset + i;
+		     i += 1) {
+			config->conversion_buf[i * 4] = buf[cur_offset + i] & BIT(1) ? 0xF : 0;
+			config->conversion_buf[i * 4] |= (buf[cur_offset + i] & BIT(0) ? 0xF : 0)
+							 << 4;
+			config->conversion_buf[i * 4 + 1] = buf[cur_offset + i] & BIT(3) ? 0xF : 0;
+			config->conversion_buf[i * 4 + 1] |=
+				(buf[cur_offset + i] & BIT(2) ? 0xF : 0) << 4;
+			config->conversion_buf[i * 4 + 2] = buf[cur_offset + i] & BIT(5) ? 0xF : 0;
+			config->conversion_buf[i * 4 + 2] |=
+				(buf[cur_offset + i] & BIT(4) ? 0xF : 0) << 4;
+			config->conversion_buf[i * 4 + 3] = buf[cur_offset + i] & BIT(7) ? 0xF : 0;
+			config->conversion_buf[i * 4 + 3] |=
+				(buf[cur_offset + i] & BIT(6) ? 0xF : 0) << 4;
+		}
+	}
+	return i;
 }
 
-static int ssd1322_write_pixels(const struct device *dev, const uint8_t *buf, uint32_t pixel_count)
+static int ssd1322_write_pixels_MONO01(const struct device *dev, const uint8_t *buf,
+				       uint32_t pixel_count,
+				       const struct display_buffer_descriptor *desc)
```

Comment:
Variables and functions typically use [snake_case](https://docs.zephyrproject.org/latest/contribute/style/code.html) like in the original function.
It does not break the driver to change the coding style though.

Reply:
this intentionally reproduces the case found in the defines for the pixel format

Reply:
Thank you for confirming. It seems like occasionally done for i.e. standards.